libsolidity/parsing/Token.cpp

```diff
@@ -80,8 +102,44 @@ char const Token::m_tokenType[] =
 {
 	TOKEN_LIST(KT, KK)
 };
-Token::Value Token::fromIdentifierOrKeyword(const std::string& _name)
+unsigned Token::extractM(string const& _literal)
+{
+	try
+	{
+		unsigned short m = stoi(_literal);
+		return m;
+	}
+	catch(out_of_range& e)
+	{
+		return 0;
+	}
+}
+tuple<Token::Value, unsigned short, unsigned short> Token::fromIdentifierOrKeyword(string const& _literal)
 {
+	auto positionM = find_if(_literal.begin(), _literal.end(), ::isdigit);
+	if (positionM != _literal.end())
+	{
+		string baseType(_literal.begin(), positionM);
+		auto positionX = find(positionM, _literal.end(), 'x');
+		unsigned short m = extractM(string(positionM, positionX));
+		if (baseType == toString(Token::Bytes))
```

Comment:
This might be weird, but you could actually call `baseToken = fromIdentifierOrKeyword(baseType)` here and compare it to `Token::Bytes` and the others.